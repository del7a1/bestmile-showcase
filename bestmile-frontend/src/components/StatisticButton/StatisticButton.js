import React, { Component } from 'react';
import axios from 'axios';
import Popup from 'react-popup'
import StatisticPopup from '../StatisticPopup/StatisticPopup'

export default class StatisticButton extends Component {

  constructor() {
    super();
    this.showStatistics = this.showStatistics.bind(this);
  }

  showStatistics() {
    axios.get('http://localhost:8080/mission/statistic')
      .then(response => {
        const data = response.data
        const missionsCounter = data.missionsCounter;
        const totalPassengerCount = data.totalPassengerCount;
        const totalDistance = data.totalDistance;
        Popup.plugins().statistics(missionsCounter, totalPassengerCount, totalDistance);
      })
      .catch((error) => {
         console.lo(error);
         Popup.alert("Can't fetch statistics from server. Please try again later");
      });
  }

  render() {
    return (
      <div>
        <button onClick={this.showStatistics}>Show Statistics</button>
        <Popup />
      </div>
    );
  }

}
