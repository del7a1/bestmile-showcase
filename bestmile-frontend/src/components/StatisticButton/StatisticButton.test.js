import React from 'react';
import { shallow } from 'enzyme';
import StatisticButton from './StatisticButton.js';

describe('StatisticButton', () => {

 it('Should be defined', () => {
   expect(StatisticButton).toBeDefined();
 });

 it('Should render correctly', () => {
   const wrapper = shallow(
     <StatisticButton />
   );
   expect(wrapper).toMatchSnapshot();
 });

 it('Should has defined value', () => {
   const wrapper = shallow(
     <StatisticButton />
   );
   expect(wrapper.find('button').text()).toEqual('Show Statistics');
 });

});
