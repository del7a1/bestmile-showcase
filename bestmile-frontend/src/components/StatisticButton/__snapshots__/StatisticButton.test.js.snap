// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatisticButton Should render correctly 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Show Statistics
  </button>
  <Popup
    btnClass="mm-popup__btn"
    className="mm-popup"
    closeBtn={true}
    closeHtml={null}
    closeOnOutsideClick={true}
    defaultCancel="Cancel"
    defaultCancelKey="esc"
    defaultOk="Ok"
    defaultOkKey="enter"
    escToClose={true}
  />
</div>
`;
